nibabel/streamlines/tests/test_tractogram.py

```diff
@@ -800,29 +816,28 @@ def test_lazy_tractogram_creation(self):
         # Creating LazyTractogram with generators is not allowed as
         # generators get exhausted and are not reusable unlike generator
         # function.
-        assert_raises(TypeError, LazyTractogram, streamlines)
-        assert_raises(TypeError, LazyTractogram,
-                      data_per_point={"none": None})
-        assert_raises(TypeError, LazyTractogram,
-                      data_per_streamline=data_per_streamline)
-        assert_raises(TypeError, LazyTractogram, DATA['streamlines'],
-                      data_per_point=data_per_point)
+        with pytest.raises(TypeError):
+            LazyTractogram(streamlines=streamlines)
+            LazyTractogram(data_per_point={"none": None})
+            LazyTractogram(data_per_streamline=data_per_streamline)
+            LazyTractogram(streamlines=DATA['streamlines'],
+                           data_per_point=data_per_point)
```

Comment:
Please add a `with pytest.raises` for each of these. If it's more appealing, you can use `pytest.raises` exactly like you would `assert_raises`, and skip the `with` blocks.